src/controller/java/src/chip/setuppayload/SetupPayloadParser.kt

```diff
@@ -0,0 +1,108 @@
+package chip.setuppayload
+
+import java.util.logging.Level
+import java.util.logging.Logger
+
+/** Parser for scanned QR code or manual entry code.  */
+class SetupPayloadParser {
```

Comment:
In spec it is called "onboarding payload".  "Setup Payload" is the legacy pre-spec SDK name.